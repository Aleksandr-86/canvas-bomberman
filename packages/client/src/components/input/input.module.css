.form__input {
  display: block;
  width: 100%;
  margin-bottom: 30px;
  font-family: 'Rubik Bubbles', fantasy;
  cursor: pointer;
}
