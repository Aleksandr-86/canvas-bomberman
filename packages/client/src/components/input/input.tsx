import classes from './input.module.css'


interface Props {
    placeholder: string
    class?: string
    type: string
}

export const Input = (props: Props) => {
    const { placeholder = 'Инпут' } = props

    if (props.class) {
        return (
            <label>
                <input className={`${classes.form__input} ${props.class}`} type={props.type} placeholder={placeholder} />
            </label>
        )
    } else {
        return (
            <label>
                <input className={classes.form__input} type={props.type} placeholder={placeholder} />
            </label>
        )
    }
}
