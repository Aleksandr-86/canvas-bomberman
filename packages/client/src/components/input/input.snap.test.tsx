import * as renderer from 'react-test-renderer'
import { Input } from './input'

const mockProps = {
  class: `mockClass`,
  placeholder: `mockPlaceholder`,
  type: `text`,
}

describe(`<Input/> snapshot test cases`, () => {
  test(`<Input/> should render <input>`, () => {
    const tree = renderer.create(<Input {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
