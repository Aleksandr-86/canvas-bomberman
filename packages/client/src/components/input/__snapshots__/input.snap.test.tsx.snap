// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> snapshot test cases <Input/> should render <input> 1`] = `
<label>
  <input
    className="formInput"
    placeholder="mockPlaceholder"
    type="text"
  />
</label>
`;
